tensorflow_datasets/scripts/generate_visualization.py

```diff
@@ -0,0 +1,62 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Script to generate datasets figures and save them
+  in docs/catalog/images/ directory.
+
+Args:
+  data_name: name of the dataset, for which figures to be generated.
+"""
+
+import os
+from absl import app
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+FIG_DIR = os.path.join('..', 'docs', 'catalog', 'images')
+FULL_PATH = tfds.core.get_tfds_path(FIG_DIR)
+
+def generate_vizz(data_name):
+    """Save the generated figures for the dataset
+    Args:
+      data_name: name of the dataset
+    """
+    try:
+        print("Generating examples...")
+        builder = tfds.builder(data_name)
+        builder.download_and_prepare()
```

Comment:
As explained on the issue, the script should assume the datasets already exists (no call to `download_and_prepare`). Could you instead create a script like: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/scripts/document_datasets.py

Create a `def generate_visualization(datasets=None):` which use thread pool to generate the builder images in parallel.